lib/build.js

```diff
@@ -88,34 +88,55 @@ class BuildModel extends BaseModel {
             jobId,
             createTime: now,
             number: now,
-            status: 'QUEUED',
-            sha
+            status: 'QUEUED'
         };
+        let buildInfo;
         let repoInfo;
 
         async.waterfall([
-            (next) => {
+            async.apply(fetchJob, this.datastore, jobId),
+            async.apply(fetchScmUrl, this.datastore),
+
+            (data, next) => {
```

Comment:
I think this is overly complex, and could be refactored.

You can make a method that would return a sha, either from the given data, or as a result of a github lookup. This can be refactored out of the create method to reduce complexity. The following method in the waterfall which performs `datastore.save()` could then just take in a sha, and save it to `initialBuildData`.